.circleci/check_gotest.sh

```diff
@@ -2,14 +2,29 @@
 
 # Run test in each go environment and log errors to $FILEPATH_ERROR_FLAG.
 # If all tests pass successfully then the file will not be created at all.
-export GOPATH=$HOME/$GOPATH_FOLDER;
 # cd to code with go.mod file outside of GOPATH to work with module-based behaviour
 # https://github.com/golang/go/wiki/Modules#when-do-i-get-old-behavior-vs-new-module-based-behavior
 cd $HOME/project
 
-for version in $VERSIONS; do
-    export GOROOT=$HOME/go_root_$version/go;
-    export PATH=$GOROOT/bin/:$PATH;
+OLD_PATH=$PATH
+
+if [ -z "$GO_VERSIONS" ]; then
+    GO_VERSIONS="$(echo $GOROOT | sed -E 's|^/usr/lib/go/([0-9]+\.[0-9]+(\.[0-9]+)?)/go$|\1|')"
+fi
+
+for go_version in $GO_VERSIONS; do
+    export GOROOT=/usr/lib/go/$go_version/go
+
+    if [ ! -d $GOROOT ]; then
+        echo "Error: Go $go_version is not installed, $GOROOT does not exist"
+        exit 1
+    fi
+
+    export PATH=$GOROOT/bin:$OLD_PATH
+
+    echo "Using Go $go_version at $GOROOT"
+    echo PATH=$PATH
```

Comment:
Also, it might be good idea to dump `$(go version)` output to make sure that Go version is correct.